pulp_ostree/tests/functional/api/test_modify.py

```diff
@@ -1,275 +1,270 @@
+    assert ref == removed_ref
+
+    removed_commits = ostree_content_commits_api_client.list(
+        repository_version_removed=repo1.latest_version_href
     )
-    repo_refs = sorted(
-        [remove_created_key(item) for item in repo_content[OSTREE_REFS_NAME]],
-        key=lambda item: item["pulp_href"],
+    removed_commit = ostree_content_commits_api_client.read(
+        removed_commits.to_dict()["results"][0]["pulp_href"]
     )
-    repo_configs = sorted(
-        [remove_created_key(item) for item in repo_content[OSTREE_CONFIGS_NAME]],
-        key=lambda item: item["pulp_href"],
+    assert latest_commit == removed_commit
+
+    # now, remove the second commit and check whether the referenced objects were removed too
+    response = ostree_repositories_api_client.modify(
+        ostree_ostree_repository_href=repo1.pulp_href,
+        repository_add_remove_content={"remove_content_units": [second_commit.pulp_href]},
     )
-    repo_summaries = sorted(
-        [remove_created_key(item) for item in repo_content[OSTREE_SUMMARIES_NAME]],
-        key=lambda item: item["pulp_href"],
+    monitor_task(response.task)
+
+    repo1 = ostree_repositories_api_client.read(repo1.pulp_href)
+    assert repo1.latest_version_href == f"{repo1.pulp_href}versions/3/"
+
+    repository_version = ostree_repositories_versions_api_client.read(repo1.latest_version_href)
+    removed_content = repository_version.content_summary.removed
+    assert removed_content["ostree.commit"]["count"] == 1
+    assert removed_content["ostree.object"]["count"] == 3
+
+
+@pytest.mark.parallel
+def test_add_remove_obj(
+    monitor_task,
+    ostree_content_objects_api_client,
+    ostree_repositories_api_client,
+    synced_repo_version,
+):
+    """Try to modify an object (e.g., dirtree, dirmeta, ...) in the existing repository."""
+    repo_version1, _, repo1 = synced_repo_version
+    created_objs = ostree_content_objects_api_client.list(
+        repository_version_added=repo_version1.pulp_href
     )
-    return repo_commits + repo_objects + repo_refs + repo_configs + repo_summaries
+    obj = created_objs.to_dict()["results"][0]
 
+    version1 = repo1.latest_version_href
 
-def remove_created_key(dic):
-    """Remove the key `created` from the passed dictionary."""
-    return {k: v for k, v in dic.items() if k != "created"}
+    # objects should be ignored by the machinery
+    response = ostree_repositories_api_client.modify(
+        ostree_ostree_repository_href=repo1.pulp_href,
+        repository_add_remove_content={"add_content_units": [obj["pulp_href"]]},
+    )
+    monitor_task(response.task)
+    version2 = repo1.latest_version_href
+    assert version1 == version2
+
+    response = ostree_repositories_api_client.modify(
+        ostree_ostree_repository_href=repo1.pulp_href,
+        repository_add_remove_content={"remove_content_units": [obj["pulp_href"]]},
+    )
+    monitor_task(response.task)
+    version2 = repo1.latest_version_href
```

Comment:
Same here. The latest version is not automatically updated for `repo1` since it represent an older state of the repository, before running the `modify` task.